// Copyright 2020 The Loimos Project Developers.
// See the top-level LICENSE file for details.
//
// SPDX-License-Identifier: MIT

syntax = "proto3";
package loimos.proto;

message InterventionModel {
  message Trigger {
    message TriggerType_NewDailyCases {
      double trigger_on = 1;
      double trigger_off = 2;
    }

    message TriggerType_Day {
      int32 trigger_on  = 1;
      int32 trigger_off = 2;
    }

    oneof trigger_type {
      TriggerType_NewDailyCases new_daily_cases = 1;
      TriggerType_Day           day             = 2;
    }
  }

  message Intervention {
    double compliance = 1;
    
    message InterventionType_SelfIsolation {}

    message InterventionType_SchoolClosures {
      int32 csv_location_of_school = 1;
    }
    
    message InterventionType_Vaccination {
      int32  per_day = 1;
      double infection_protection = 2;
      double severe_disease_protection = 3;
      double transmission_protection = 4;
      int32  delay = 5;
    }
    
    oneof intervention_type {
      InterventionType_SelfIsolation  self_isolation  = 2;
      InterventionType_SchoolClosures school_closures = 3;
      InterventionType_Vaccination    vaccination     = 4;
    }
  }
  
  Trigger trigger = 1;
  optional Intervention self_isolation  = 2;
  optional Intervention school_closures = 3;
  optional Intervention vaccination     = 4;
  
  // CSV Mappings
  int32 csv_location_of_seeder_bool = 5;
  int32 seeding_admin_code = 6;
}
